src/modules/claims/claims.service.ts

```diff
@@ -1250,21 +1250,21 @@ export class ClaimsService {
     const { enrolmentPreconditions } = roleDefinition as IRoleDefinition;
```

Comment:
```suggestion
    const { enrolmentPreconditions } = roleDefinition as IRoleDefinitionV2;
```
@whitneypurdum The IRoleDefinitionV2 is the interface with `revokers` and `defaultValidityPeriod` https://github.com/energywebfoundation/ew-credentials/blob/77bd399bbbe78e32f85a4884e4e7073bdeff0dee/packages/credential-governance/src/types/domain-definitions.ts#L23

Reply:
I think we should do a type check, as the roleDefinition coming from `this._domainsService.getDefinition` returns `Promise< IRoleDefinition | IRoleDefinitionV2 | IAppDefinition | IOrganizationDefinition`. https://github.com/energywebfoundation/iam-client-lib/blob/ff6886c517bc55195691d1d316b770afa4dffde7/src/modules/domains/domains.service.ts#L802

Reply:
@jrhender @JGiter We should create separate methods to get definitions, fetching everything with one function `domainsService.getDefinition` can be confusing and would need type validation every time. What are your thoughts?

Reply:
Make sense to me

Reply:
Thanks @nichonien I updated that, I think I just need enrolmentPreconditions, which is on both interfaces though.